base/multimedia.jl

```diff
@@ -249,39 +249,79 @@ close(d::TextDisplay) = close(d.io)
 # We keep a stack of Displays, and calling display(x) uses the topmost
 # AbstractDisplay that is capable of displaying x (doesn't throw an error)
 
+# this stays as is for now and represents all displays of priority 0 for
+# backwards compatibility
 const displays = AbstractDisplay[]
+const display_stack = IdDict{Int, Vector{AbstractDisplay}}()
```

Comment:
Maybe a vector of `(priority, display)` sorted by priority would be easier to use here. Inserting and removing displays is rare, and then each call to `display` wouldn't need to call `sort`.

Reply:
Good point -- punting the sorting to `pushdisplay`/`popdisplay` is much better.